### PR TITLE
Update workflows

### LINKED ISSUE #1922: AutoRev is not picking up v3.1.x releases on master branch
**Describe the bug**
AutoRev is not picking up v3.1.x releases on `master` branch

**To Reproduce**
1. Tag a release on `master` branch
2. Notice that the compiled binaries show incorrect version.

https://github.com/longturn/freeciv21/actions/runs/4779208059/jobs/8496507906

**Expected behavior**
AutoRev properly grabs the tag

**Screenshots**
image

image

**Platform and version (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Freeciv21 version: v3.1-alpha.1
 - Ruleset/Longturn game (if applicable): [e.g. `classic` or `LTEx24`]

**Additional context**

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1922

### CODE REVIEW

**Comment by jwrober** [comment]
Two passes of release on my fork seem to work, however I don't think that this is 100% ready. @lmoureaux thoughts?

**Comment by lmoureaux** [comment]
What does the `workflow_run` block do?

**Comment by jwrober** [comment]
> What does the `workflow_run` block do?

Its supposed to act as a "uses" so release should wait on build. That doesn't seem to do anything. I'm leaning to the change in checkout

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ on:
   schedule:
     - cron: "0 5 * * MON"
   workflow_dispatch: {}
+  workflow_run:
+    workflows: ["Build"]
+    types:
+      - completed
```
Comment (lmoureaux) [inline .github/workflows/release.yaml:11]: Let's remove it if it doesn't work
```suggestion
```

**Review by lmoureaux — CHANGES_REQUESTED** [review]

**Review by lmoureaux — APPROVED** [review]

**Pull request merged**